amaranth/hdl/ast.py

```diff
@@ -156,6 +156,23 @@ def signed(width):
     return Shape(width, signed=True)
 
 
+def overridable_by_reflected(operator):
```

Comment:
This should not be a public function as the RFC did not include a public API addition.

Reply:
Re: the suggestion to put it in `_utils`, you could take `ValueCastable` by argument of decorator.

Reply:
I just made an attempt at this. It runs into the issue that `ValueCastable` doesn't yet exist when the decorator is called, since it's defined after `Value`.

Reply:
Ah right. The current location is fine then.